components/tikv_util/src/waitgroup.rs

```diff
@@ -0,0 +1,216 @@
+// Copyright 2023 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::{
+    pin::Pin,
+    sync::{
+        atomic::{AtomicBool, AtomicUsize, Ordering},
+        Arc,
+    },
+};
+
+use futures::future::{Future, FutureExt};
+use tokio::sync::oneshot;
+
+pub type BoxFuture<T> = Pin<Box<dyn Future<Output = T> + Send>>;
+
+pub struct WaitGroup {
```

Comment:
why not use https://docs.rs/crossbeam/latest/crossbeam/sync/struct.WaitGroup.html

Reply:
This is because crossbeam::WaitGroup is not async...I did not find a more suitable

Reply:
how about this https://docs.rs/waitgroup/latest/waitgroup/

Reply:
because I need errgroup, I will try to use that. What do u think? :)
https://docs.rs/futures-util/latest/futures_util/future/fn.try_join_all.html

Reply:
done.